paywall/src/unlock.js/setupPostOffices.js

```diff
@@ -35,6 +39,55 @@ export default function setupPostOffices(window, dataIframe, CheckoutUIIframe) {
     process.env.PAYWALL_URL
   )
 
+  if (!loadCheckoutModal) {
+    loadCheckoutModal = () => {
+      showIframe(window, CheckoutUIIframe)
+    }
+  }
+  const hideCheckoutModal = () => {
+    hideIframe(window, CheckoutUIIframe)
+  }
+
+  const unlockProtocol = {}
+
+  Object.defineProperties(unlockProtocol, {
+    loadCheckoutModal: {
+      value: loadCheckoutModal,
+      writable: false, // prevent changing loadCheckoutModal by simple `unlockProtocol.loadCheckoutModal = () => {}`
+      configurable: false, // prevent re-defining the writable property
+      enumerable: false, // prevent finding it exists via `for ... of`
+    },
+  })
+
+  const freeze = Object.freeze || Object
```

Comment:
I believe the freezing stuff here would benefit from being extracted into its own method